### ISSUE TITLE
Framedrops in games for couple of seconds during injection

### ISSUE THREAD

**Issue by saikat0511** [sezanzeb/key-mapper#128]
I am using a python script along with key-mapper to automatically switch presets when windows lose/gain focus. 

While the script works as expected, but when I switch to the game window the cursor stops working (understandable because its injecting, that's not the problem) and the game drops frames for couple of seconds before going back to normal.

I have tried checking the system monitor while switching windows and what I have noticed is that the process Xorg jumps to about 11% CPU usage when the frame drops begin. So that might be the cause but I am not sure...

Tbh its not that big of a issue and I can definitely live with it, but still it would be great if this can get fixed somehow

Also this is the py script I am using to switch presets. It just uses xdotool and subprocess to call key-mapper-control so I doubt this would cause something like that. Sharing it here just in case...
[switch.py.txt](https://github.com/sezanzeb/key-mapper/files/6693193/switch.py.txt)

**Comment by sezanzeb** [comment]
possibly related to https://github.com/sezanzeb/key-mapper/issues/89

I guess calling key-mapper-control manually without xdotool also causes framedrops?

**Comment by sezanzeb** [comment]
While I would love to know what causes framdropds, I can't reproduce it and I also don't know how to profile X11 to find the calls that use up the cpu.

**Comment by saikat0511** [comment]
> Same problem with wayland?

xdotool doesn't work in wayland so.....

**Comment by sezanzeb** [comment]
yeah, but if you trigger key-mapper-control manually, do you also get framedrops in wayland?

**Comment by saikat0511** [comment]
wine doesn't support wayland so cant test that either because all my games use wine+dxvk. 

I will try to test it with some native game later

**Comment by saikat0511** [comment]
After further testing I noticed that I get screen freezes when pressing the mouse extra buttons too (the ones mapped with key-mapper). What's interesting is that when there are no lags in the OBS recordings

Nvidia driver 470 allowed vulkan under wayland so I tested my games with games on a wayland setup and there were no lags related to key-mapper whatsoever. It seems this issue is exclusive to x11

**Comment by saikat0511** [comment]
okay some more info.... There are no issues when running games in fullscreen mode in xorg. But in windowed mode there are lags as described in my earlier posts.
